File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/annotations/PermissionContainer.java
package net.crystalapps.permission.runtime.library.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface PermissionContainer {
    Permit[] value();
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/annotations/PermissionContainer.kt
package net.crystalapps.permission.runtime.library.annotations


/**
 * Created by Syed Owais Ali on 10/13/2018.
 */

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class PermissionContainer(vararg val value: Permit)
